### PR TITLE
Improve documentation of numpy.ma.filled

### LINKED ISSUE #9716: DOC: Potential Copy Pitfall in `numpy.ma.filled`
Pull request [#6920][pull request] added a note to `ma.MaskedArray.filled` explaining when a copy is returned by that method. Should a similar note be added to [`ma.filled`][filled definition], which calls `ma.MaskedArray.filled`? The documentation there neither mentions the dependence on the input's mask nor links to `ma.MaskedArray.filled`, so the only way to learn about the behavior is to read the source.

[pull request]: https://github.com/numpy/numpy/pull/6920
[filled definition]: https://github.com/numpy/numpy/blob/master/numpy/ma/core.py#L590

### PR DESCRIPTION
**Pull request by karthikkaiplody**
Closes #9716

### CODE REVIEW

File: numpy/ma/core.py [hunk 1]
```diff
@@ -608,7 +608,9 @@ def common_fill_value(a, b):
 
 def filled(a, fill_value=None):
     """
-    Return input as an array with masked data replaced by a fill value.
+    Return a copy of `a`, with masked values replaced by a fill value.
+    However, if there are no masked values to fill, `a` will be returned
+    instead as an ndarray.
```
Comment (mattip) [inline numpy/ma/core.py:613]: Is it "`a`'s data will be returned" or "a copy of `a` will be returned"?

Comment (seberg) [reply numpy/ma/core.py:613]: Also note that this information shouldn't be part of the initial sentence, but rather the following paragraph.

**Review by rossbar — COMMENTED** [review]
Since they were minor changes, I took the liberty of incorporating the review feedback and pushing up the tweaks.

**Comment by mattip** [comment]
Thanks @karthikkaiplody, @rossbar

**Pull request merged**